src/main/java/org/example/PetShelterAdd.java

```diff
@@ -1,4 +1,200 @@
+    }
+
+    private void showAllPets() {
+        if (animals.isEmpty()) {
+            System.out.println("There are no pets in the shelter.");
+        } else {
+            System.out.println("Pets in the shelter:");
+            for (Animal pet : animals) {
+                System.out.println(pet);
+            }
+        }
+    }
+
+    private void removePetFromShelter() {
+        Scanner scanner = new Scanner(System.in);
+
+        if (animals.isEmpty()) {
+            System.out.println("The shelter has no pets to remove.");
+            return;
+        }
+
+        System.out.println("Select a pet to delete:");
+
+        for (int i = 0; i < animals.size(); i++) {
+            System.out.println((i + 1) + ". " + animals.get(i));
+        }
+
+        System.out.print("Enter the pet number to delete:");
+        int choice = scanner.nextInt();
+
+        if (choice > 0 && choice <= animals.size()) {
+            Animal removedPet = animals.remove(choice - 1);
+            System.out.println("Pet removed: " + removedPet);
+        } else {
+            System.out.println("Incorrect selection of pet to delete.");
+        }
+    }
+
+    public void run() {
+        Scanner scanner = new Scanner(System.in);
+
+        System.out.println("Hello! Welcome to the Pet Shelter. What is your name?");
+        String name = scanner.nextLine();
+        System.out.println("Your name: " + name);
+        System.out.println("Nice to meet you, " + name + "! On our website, in the PHOTO section, you can see all our FLUFFIES " +
+                "and choose who you are looking for.");
+
+        boolean running = true;
+        while (running) {
+            System.out.println("Menu:");
+            System.out.println("1. Add a pet");
+            System.out.println("2. View all pets");
+            System.out.println("3. Remove a pet");
+            System.out.println("4. Exit");
+
+            System.out.print("Select an option:");
+            if (!scanner.hasNextInt()) {
+                System.out.println("Incorrect choice. Please enter a number.");
+                scanner.nextLine(); // discard the incorrect input
+                continue;
+            }
+
+            int choice = scanner.nextInt();
+
+            switch (choice) {
+                case 1:
+                    addPet();
+                    break;
+                case 2:
+                    showAllPets();
+                    break;
+                case 3:
+                    removePetFromShelter();
+                    break;
+                case 4:
+                    saveAnimalsToFile();
+                    System.out.println("Thank you for using. Goodbye, " + name + "!");
+                    running = false;
+                    break;
+                default:
+                    System.out.println("Incorrect choice. Try again.");
+            }
+        }
+    }
+
+    public void saveAnimalsToFileForTest() {
```

Comment:
Try to avoid such approaches, it's better to use real public method to test it or consider using reflection